clang/lib/DPCT/ExprAnalysis.cpp

```diff
@@ -657,12 +657,26 @@ void ExprAnalysis::analyzeExpr(const CXXConstructExpr *Ctor) {
 }
 
 void ExprAnalysis::analyzeExpr(const MemberExpr *ME) {
-  auto PP = DpctGlobalInfo::getContext().getPrintingPolicy();
-  PP.PrintCanonicalTypes = true;
-  auto QT = ME->getBase()->getType();
-  auto BaseType = (QT->isPointerType() ? QT->getPointeeType() : QT)
-                      .getUnqualifiedType()
-                      .getAsString(PP);
+  auto BaseType = [&]() {
```

Comment:
Why we need wrapping code in the lambda here?

Reply:
The code within the lambda is all logic for initializing `BaseType`. Without using a lambda, we would have a declaration `std::string BaseType;` and the lambda body inline with `BaseType = <expr>` instead of `return <expr>`. Both do the same thing, but the lambda method makes it more clear (in my opinion) that the code inside that the code inside lambda body is only for initializing `BaseType`. In fact, we can say `const auto BaseType = ...` with the lambda method - without the lambda method, one cannot say `const std::string;`. This method of complex initialization is recommended by the [C++ core guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es28-use-lambdas-for-complex-initialization-especially-of-const-variables).